bazel/canbench.bzl

```diff
@@ -26,7 +26,7 @@ def rust_canbench(name, results_file, add_test = False, **kwargs):
     wasm_rust_binary_rule(
         name = name + "_wasm",
         binary = ":{name}_bin".format(name = name),
-        opt = "3",
+        opt = "s",
```

Comment:
The [default optimisation level for rust canisters is 3](https://sourcegraph.com/github.com/dfinity/ic/-/blob/bazel/canisters.bzl?L61). Shouldn't `rust_canbench` benchmark canisters as they are in production, so with level 3?

Also, if "unrelated" changes can effect the performance of a canister. Shouldn't `rust_canbench` detect such cases?

What's an example of an unrelated change that effects the performance?

Reply:
As also mentioned on the internal slack thread:

I would propose the following:

* Remove the [the WASM target](https://sourcegraph.com/github.com/dfinity/ic/-/blob/bazel/canbench.bzl?L21-L30) from `rust_canbench` and pass it in as a parameter instead.
* For [all invocations of rust_canbench](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/dfinity/ic%24+rust_canbench%28&patternType=keyword&sm=0) apply it to the production canister except for the NNS governance canister
* Add a special governance canister target for canbench with the `s` optimisation level and add a `rust_canbench` target for that.

This has the additional benefit that we're not building canisters multiple times and you're being explicit that you're not benchmarking the production NNS governance canister but a modified version.
CC @gregorydemay for the FI canbenches.